conjure-go-client/httpclient/internal/request_retrier.go

```diff
@@ -45,11 +45,14 @@ type RequestRetrier struct {
 // NewRequestRetrier creates a new request retrier.
 // Regardless of maxAttempts, mesh URIs will never be retried.
 func NewRequestRetrier(uris []string, retrier retry.Retrier, maxAttempts int) *RequestRetrier {
-	offset := rand.Intn(len(uris))
+	offset := 0
+	urisCopy := make([]string, len(uris))
+	copy(urisCopy, uris)
+	rand.Shuffle(len(urisCopy), func(i, j int) { urisCopy[i], urisCopy[j] = urisCopy[j], urisCopy[i] })
```

Comment:
note: not for this PR, but I would like to remove using the global rand here in place of either a user provided one or setup a local seed so we can have true randomness.